tests/azure_iot_hub_client_ut.c

```diff
@@ -0,0 +1,351 @@
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Connect, eAzureIoTMQTTServerRefused );
+    assert_int_not_equal( AzureIoTHubClient_Connect( &xTestIoTHubClient,
+                                                     false,
+                                                     &xSessionPresent,
+                                                     60 ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Connect_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    bool xSessionPresent;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Connect, eAzureIoTMQTTSuccess );
+    assert_int_equal( AzureIoTHubClient_Connect( &xTestIoTHubClient,
+                                                 false,
+                                                 &xSessionPresent,
+                                                 60 ),
+                      eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Disconnect_InvalidArgFailure( void ** state )
+{
+    ( void ) state;
+
+    assert_int_not_equal( AzureIoTHubClient_Disconnect( NULL ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Disconnect_MQTTDisconnectFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Disconnect, eAzureIoTMQTTIllegalState );
+    assert_int_not_equal( AzureIoTHubClient_Disconnect( &xTestIoTHubClient ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Disconnect_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Disconnect, eAzureIoTMQTTSuccess );
+    assert_int_equal( AzureIoTHubClient_Disconnect( &xTestIoTHubClient ),
+                      eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_SendTelemetry_InvalidArgFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    /* NULL handler */
+    assert_int_not_equal( AzureIoTHubClient_SendTelemetry( NULL,
+                                                           ucTestTelemetryPayload,
+                                                           sizeof( ucTestTelemetryPayload ) - 1, NULL ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_SendTelemetry_BigTopicFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+    AzureIoTMessageProperties_t properties =
+    {
+        ._internal.xProperties._internal.current_property_index = 0,
+        ._internal.xProperties._internal.properties_written     = sizeof( ucBuffer ) + 1
```

Comment:
>     [](start = 60, length = 4)

nit: remove extra space

Reply:
this is using uncrustify so I'll just let it run its course